content/blog/pairisto/2023-09-10_[gsoc_2023]_vpn_support_project_final_report.md

```diff
@@ -0,0 +1,42 @@
++++
+type = "blog"
+title = "[GSoC 2023] VPN Support Project Final Report"
+author = "Pairisto"
+date = "2023-09-10 23:29:07-05:00"
+tags = ["haiku", "software"]
++++
+
+## Project Overview
+This project, undertaken as part of Google Summer of Code 2023, sets out to implement a robust TUN/TAP driver for Haiku given the increasing demand from the community for a virtual network kernel interface. This project allows for VPN software and other network-related utilities that work with TUN/TAP driver to operate seamlessly on Haiku. Throughout the project's duration, I've documented the rationale behind key design decisions, and offered insights on the interplay between the TUN/TAP driver and Haiku's unique networking architecture in my series of seven progress blog posts. For those interested in trying out the TUN/TAP driver on Haiku (when the code is merged, more on that in a different section), you can use the port of OpenVPN that was made on haikuports [here](https://github.com/haikuports/haikuports/tree/master/net-vpn/openvpn). One last thing to say is that I could not get the TUN interface to work properly due to a lack of Point-to-Point being supported by Haiku but TAP works for both the interface and driver.
+
+## What Was Done
+1. Added the TUN/TAP driver and interface are in a state that works for tunneled networking communications over TAP (explained why TUN can't be used in the next section).
+2. OpenVPN was ported over and modified to work on Haiku to work with the TUN/TAP driver created.
+
+## What's Left To Do
+1. Add asynchronous I/O support to the operating system to support AIO functionality that some applications might work off of.
```

Comment:
Which applications? IMO there are only a few applications using AIO.

Reply:
there are a few, yes, but all other TUN/TAP drivers have this functionality as an option. I think that this is something that we should add later as an option since someone might want to port something over that uses AIO.

Reply:
Please name some applications, otherwise that's a moot point. We already have a bug report https://dev.haiku-os.org/ticket/5683  and until now it's just to comply with the POSIX spec.

Reply:
Changed